services/business/checkers/virtual_services/route_checker.go

```diff
@@ -0,0 +1,97 @@
+package virtual_services
+
+import (
+	"reflect"
+	"strconv"
+
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/services/models"
+	"github.com/kiali/kiali/util/intutil"
+)
+
+type RouteChecker struct{ kubernetes.IstioObject }
+
+// Check returns both an array of IstioCheck and a boolean indicating if the current route rule is valid.
+// The array of IstioChecks contains the result of running the following validations:
+// 1. All weights with a numeric number.
+// 2. All weights have value between 0 and 100.
+// 3. Sum of all weights are 100 (if only one weight, then it assumes that is 100).
+// 4. All the route has to have weight label.
+func (route RouteChecker) Check() ([]*models.IstioCheck, bool) {
+	checks, valid := route.checkRoutesFor("http")
+	tcpChecks, tcpValid := route.checkRoutesFor("tcp")
+	return append(checks, tcpChecks...), valid && tcpValid
+}
+
+func (route RouteChecker) checkRoutesFor(kind string) ([]*models.IstioCheck, bool) {
+	validations := make([]*models.IstioCheck, 0)
+
+	var weightSum int
+	var weightCount int
+	var valid = true
+
+	http := route.GetSpec()[kind]
+	if http == nil {
+		return validations, valid
+	}
+
+	// Getting a []HTTPRoute
+	slice := reflect.ValueOf(http)
+	if slice.Kind() != reflect.Slice {
+		return validations, valid
+	}
+
+	for i := 0; i < slice.Len(); i++ {
+		route, ok := slice.Index(i).Interface().(map[string]interface{})
+		if !ok || route["route"] == nil {
+			continue
+		}
+
+		// Getting a []DestinationWeight
+		destinationWeights := reflect.ValueOf(route["route"])
+		if destinationWeights.Kind() != reflect.Slice {
+			return validations, valid
+		}
+
+		for j := 0; j < destinationWeights.Len(); j++ {
+			destinationWeight, ok := destinationWeights.Index(j).Interface().(map[string]interface{})
+			if !ok || destinationWeight["weight"] == nil {
+				continue
+			}
+
+			weightCount = weightCount + 1
```

Comment:
It could be nice if weightCount is initialized at the begining of the declaration.
By default is 0 and it's fine, but I guess it helps to read the logic of the loop adding an explicit 0.
(disclaimer, I have no idea what is the normal way in go)

Reply:
Actually, this is a bug. WeightCount and weightSum should bet set to 0 in each iteration. Good.